server.js

```diff
@@ -25,41 +29,59 @@ app.get('/', (req, res) => {
 });
 
 app.get('/location', (req, res) => {
-  if (req.query.city === '') {
-    res.status(500).send('No Data My Bad.');
-    return;
-  }
-
   const searchLocation = req.query.city;
-  const locationDataUrl = `https://us1.locationiq.com/v1/search.php?key=${locationIqKey}&q=${searchLocation}&format=json`;
-  superagent.get(locationDataUrl).then(locationDataReturn => {
-
-    const locationData = locationDataReturn.body[0];
-    const newLocation = new Location(
-      searchLocation,
-      locationData.display_name,
-      locationData.lat,
-      locationData.lon
-    );
-    res.send(newLocation);
-  })
-    .catch(error => {
-      res.status(500).send('locationiq failed');
-      console.log(error.message);
-    });
+  const locationIqKey = process.env.locationIqKey;
+
+  const sqlQuery = 'SELECT * FROM place WHERE search_query=$1';
+  const sqlArray = [searchLocation];
+
+  client.query(sqlQuery, sqlArray).then(result => {
+    console.log(result.rows);
+
+    if (result.rows.length === 0) {
+
+      if (req.query.city === '') {
+        res.status(500).send('No Data My Bad.');
+        return;
+      }
+
+      const locationDataUrl = `https://us1.locationiq.com/v1/search.php?key=${locationIqKey}&q=${searchLocation}&format=json`;
+      superagent.get(locationDataUrl).then(locationDataReturn => {
+
+        const locationData = locationDataReturn.body[0];
+
+        const newLocation = new Location(
+          searchLocation,
+          locationData.display_name,
+          locationData.lat,
+          locationData.lon
+        );
+
+        const sqlQuery = 'INSERT INTO place (search_query, formatted_query, latitude, longitude) VALUES($1, $2, $3, $4)';
+        const sqlArray = [newLocation.search_query, newLocation.formatted_query, newLocation.latitude, newLocation.longitude];
+
+        client.query(sqlQuery, sqlArray);
+
+        res.send(newLocation);
+      })
+        .catch(error => {
+          res.status(500).send('locationiq failed');
+          console.log(error.message);
+        });
+    }
+  });
+
```

Comment:
Great job using intuitive variable names; helps in following along with your code